tests/test_parser.py

```diff
@@ -257,6 +261,108 @@ def test_convert_to_date_cast_result_to_date(self, mocked_locale,
                          expected_output)
 
 
+class TestBsSafe(unittest.TestCase):
+    def test_bs_safeget_return_expected_text(self):
+        mocked_soup = MagicMock()
+        selector = 'some_tag'
+        mocked_soup.select.return_value = ['some text']
+        self.assertEqual(bs_safeget(mocked_soup, selector), 'some text')
+
+    def test_bs_safeget_return_false_when_selector_is_missing(
+            self):
+        mocked_soup = Mock()
+        mocked_soup.select.return_value = ''
+        selector = 'h1'
+        self.assertFalse(bs_safeget(mocked_soup, selector))
+
+
+@patch('smarsy.parse.bs_safeget')
+class TestCreateParentsDict(unittest.TestCase):
+    def test_get_parents_img_return_expected_img(self, mocked_get):
+        mocked_get.return_value = 'some_img.jpg'
+        expected = 'some_img.jpg'
+        actual = get_parents_img('html', 'tag')
+        self.assertEqual(actual, expected)
+
+    def test_get_parents_img_return_false_with_none_img(self, mocked_get):
+        mocked_get.return_value = ''
+        self.assertFalse(get_parents_img('html', 'tag'))
+
+    def test_get_parents_name_return_expected_name(self, mocked_get):
+        mocked_get.return_value = 'name surname middlename type'
+        expected = 'name surname middlename type'
+        actual = get_parents_name('html', 'tag')
+        self.assertEqual(actual, expected)
+
+    def test_get_parents_name_return_false_with_none_name(self, mocked_get):
+        mocked_get.return_value = None
+        self.assertFalse(get_parents_name('html', 'tag'))
+
+    def test_get_parents_b_date_return_expected_b_date(self, mocked_get):
+        mocked_get.return_value = '1983-04-30'
+        expected = '1983-04-30'
+        actual = get_parents_b_date('html', 'tag')
+        self.assertEqual(actual, expected)
+
+    def test_get_parents_b_date_return_false_with_none_b_date(
+            self, mocked_get):
+        mocked_get.return_value = 0
+        self.assertFalse(get_parents_b_date('html', 'tag'))
+
+
+class TestParseParentPage(unittest.TestCase):
+    @patch('smarsy.parse.BeautifulSoup')
+    def test_parent_page_content_called_with_expected_html(self, mocked_soup):
+        html = '<tr></tr>'
+        parent_page_content_to_object(html)
+        mocked_soup.assert_called_with(html, 'html.parser')
+
+    @patch('bs4.BeautifulSoup')
+    def test_parent_page_content_raise_exeption_with_wrong_file_format(
+            self, mocked_soup):
+        html = 12344
+        self.assertRaises(TypeError, parent_page_content_to_object, html)
+
+    @patch('bs4.BeautifulSoup.select', return_value='')
+    def test_parent_page_content_return_false_with_wrong_parent_tab(
+            self, mocked_soup_select):
+        html = '<tr></tr>'
+        parent_page_content_to_object(html)
+        self.assertFalse(parent_page_content_to_object(html))
+
+    def test_parent_page_content_return_empty_list_with_no_data(self):
```

Comment:
by taking another look, this test is not good anyway:
* you should mock both BS and bs_safeget
* you should test that parent_tab was empty
In other words you'll need to return None with mock to the parent_tab and then check it. Even if result of function execution would be the same and both versions would work, logically they were doing different things.

Reply:
Refactored